src/testRunner/unittests/evaluation/usingDeclarations.ts

```diff
@@ -1810,4 +1810,48 @@ describe("unittests:: evaluation:: usingDeclarations", () => {
             "exit",
         ]);
     });
+
+    it("deterministic collapse of Await", async () => {
+        const { main, output } = evaluator.evaluateTypeScript(
+            `
+        export const output: any[] = [];
+
+        let asyncId = 0;
+        function increment() { asyncId++; }
+
+        export async function main() {
+            // increment asyncId at the top of each turn of the microtask queue
+            let pending = Promise.resolve();
+            for (let i = 0; i < 10; i++) {
+                pending = pending.then(increment);
+            }
+
+            {
+                using sync1 = { [Symbol.dispose]() { output.push(asyncId); } }; // asyncId: 2
+                await using async1 = null, async2 = null;
+                using sync2 = { [Symbol.dispose]() { output.push(asyncId); } }; // asyncId: 1
+                await using async3 = null, async4 = null;
+                output.push(asyncId); // asyncId: 0
+            }
+
+            output.push(asyncId); // asyncId: Ideally, 2, but ends up being 4 due to delays imposed by 'await'
+
+            await pending; // wait for the remaining 'increment' frames to complete.
+        }
+
+        `,
+            { target: ts.ScriptTarget.ES2018 },
+        );
+
+        await main();
+
+        assert.deepEqual(output, [
+            0,
+            1,
+            2,
+            4, // This really should be 2, but our transpile introduces an extra `await` by necessity to observe the
+               // result of DisposeResources. The process of adopting the result of DisposeResource ends up taking two
+               // turns of the microtask queue.

```

Comment:
```suggestion
            // This really should be 2, but our transpile introduces an extra `await` by necessity to observe the
            // result of DisposeResources. The process of adopting the result of DisposeResource ends up taking two
            // turns of the microtask queue.
            4,
```

Formatter's not a fan of this one being offset, it seems.